server.js

```diff
@@ -38,168 +43,178 @@ const tracks = {
-						return;
-					}
-					let forkSupported = m[1];
-
-					out = `RELEASE: ${commit}/${track}/${branch}/${forkSupported}`;
-					console.log(`Fork supported: ${forkSupported}`);
-
-					request.get({headers: { 'User-Agent': githubRepo }, url: `https://raw.githubusercontent.com/${githubRepo}/${commit}/Cargo.toml`}, function (error, response, body) {
-						if (error) throw error;
-						let version = body.match(/version = "([0-9]+)\.([0-9]+)\.([0-9]+)"/).slice(1);
-						let semver = +version[0] * 65536 + +version[1] * 256 + +version[2];
-
-						console.log(`Version: ${version.join('.')} = ${semver}`);
-
-						api.parity.registryAddress().then(a => {
-							console.log(`Registry address: ${a}`);
-							let registry = api.newContract(RegistrarABI, a);
-							return registry.instance.getAddress.call(
-								{},
-								[operationsContract, 'A']
-							);
-						}).then(a => {
-							console.log(`Parity operations address: ${a}`);
-							console.log(`Registering release: 0x000000000000000000000000${commit}, ${forkSupported}, ${tracks[track]}, ${semver}, ${isCritical}`);
-							// Should be this...
-							// api.newContract(OperationsABI, a).instance.addRelease.postTransaction({from: account.address}, [`0x000000000000000000000000${commit}`, forkSupported, tracks[track], semver, isCritical])
-							// ...but will have to be this for now...
-							return sendTransaction(OperationsABI, a, 'addRelease', [`0x000000000000000000000000${commit}`, forkSupported, tracks[track], semver, isCritical]);
-						}).then(h => {
-							console.log(`Transaction sent with hash: ${h}`);
-						});
-					});
-				});
-			}
-		});
+	const miscBody = await fetchFile(commit, '/util/src/misc.rs');
+	const branch = miscBody.match(`const THIS_TRACK. ..static str = "([a-z]*)";`)[1];
+	const track = tracks[branch] ? branch : 'testing';
+	console.log(`Track: ${branch} => ${track} (${tracks[track]}) [enabled: ${enabledTracks[track]}]`);
+
+	if (!enabledTracks[track]) {
+		throw new Error(`Track not enabled: ${track}`);
 	}
-	res.end(out);
-});
 
-app.post('/push-build/:tag/:platform', function (req, res) {
-	if (keccak256(req.body.secret) !== secretHash) {
-		res.end('Bad request.');
-		return;
+	let ethereumMod = await fetchFile(commit, '/ethcore/src/ethereum/mod.rs');
+	const network = await getNetwork();
+	const pattern = `pub const FORK_SUPPORTED_${network.toUpperCase()}: u64 = (\\d+);`;
+	const m = ethereumMod.match(pattern);
+	if (m === null) {
+		throw new Error(`Unable to detect supported fork with pattern: ${pattern}.`);
+	}
+
+	const forkSupported = m[1];
+
+	// Return a response already.
+	res.send(`RELEASE: ${commit}/${track}/${branch}/${forkSupported}`);
+
+	console.log(`Fork supported: ${forkSupported}`);
+
+	let cargoToml = await fetchFile(commit, 'Cargo.toml');
+	const version = cargoToml.match(/version = "([0-9]+)\.([0-9]+)\.([0-9]+)"/).slice(1);
+	const semver = +version[0] * 65536 + +version[1] * 256 + +version[2];
```

Comment:
Using `parseInt` might be more explicit?

Reply:
Agree.